eZ/Bundle/EzPublishCoreBundle/EventListener/SessionInitByPostListener.php

```diff
@@ -27,7 +27,10 @@ public static function getSubscribedEvents()
     public function onSiteAccessMatch(PostSiteAccessMatchEvent $event)
     {
         $request = $event->getRequest();
-        $session = $request->getSession();
+        $session = null;
+        if ($request->hasSession()) {
+            $session = $request->getSession();
+        }
 
         if (!$session || $event->getRequestType() !== HttpKernelInterface::MAIN_REQUEST) {
```

Comment:
After the change `!$session` will never happen anymore.

Reply:
@alongosz: yes, it will... If session hasn't be created, `$session` will be `null` and `!$session` will be `true`

Reply:
> @alongosz: yes, it will... If session hasn't be created, `$session` will be `null` and `!$session` will be `true`

Ok, so let's be strict about it:
```suggestion
        if (null === $session || $event->getRequestType() !== HttpKernelInterface::MAIN_REQUEST) {
```